pages/manage/productions/index.test.js

```diff
@@ -0,0 +1,58 @@
+import Index from './Index';
+import Vue from 'vue';
+
+import { mountWithEnvironment } from '../../../test/mountWithEnvironment';
+
+test('returns full list when entering the page', async () => {
+  const wrapper = mountWithEnvironment(Index);
+
+  await Vue.nextTick();
+
+  const listItems = wrapper.findAll('.productions .list-group-item');
+  expect(listItems).toHaveLength(2);
+});
+
+test('returns a found item when searching with an existing production name', async () => {
+  const wrapper = mountWithEnvironment(Index);
+
+  const searchInput = wrapper.find('#productions-overview-search');
+
+  await searchInput.setValue('Testing');
+  await Vue.nextTick();
+
+  const listItems = wrapper.findAll('.productions .list-group-item');
+  expect(listItems).toHaveLength(1);
+});
+
+test('returns no results when searching a wrong production name', async () => {
+  const wrapper = mountWithEnvironment(Index);
+
+  const searchInput = wrapper.find('#productions-overview-search');
+
+  await searchInput.setValue('test');
+  await Vue.nextTick();
+
+  const productionsContainer = wrapper.find('.productions-events-container');
+
+  const listItems = wrapper.findAll('.productions .list-group-item');
+  expect(listItems).toHaveLength(0);
+
+  expect(productionsContainer.text()).toContain(
+    'productions.overview.no_productions',
+  );
+});
+
+test('Enable remove button when selecting event', async () => {
+  const wrapper = mountWithEnvironment(Index);
+
+  await Vue.nextTick();
+  await Vue.nextTick();
```

Comment:
I may lack some knowledge about these testing tools, but it's a bit unclear to me why we need two `await Vue.nextTick()`'s here instead of the usual one in other places.

Reply:
We need to wait for vue to update the ui, so we have to use `Vue.nextTick()`. In this example we noticed it only renders fully after using `Vue.nextTick()` 2x.

Reply:
Okay thanks for the explanation. I assumed `Vue.nextTick()` waited for the next render yes. Weird that we have to do it twice here but okay, at this point the Vue magic doesn't surprise me anymore 😉 As long as it works consistently and doesn't fail every so often because it needs a _third_ `nextTick` it's fine by me 👍